boa/src/value/mod.rs

```diff
@@ -268,6 +243,37 @@ impl Value {
         }
     }
 
+    /// Converts an object to JSON, checking for reference cycles and throwing a TypeError if one is found
+    fn object_to_json(&self, obj: &GcObject, interpreter: &mut Context) -> Result<JSONValue> {
```

Comment:
I think maybe this method should instead be an associated function of the `JSON` object, the `obj` is enough to get it's fields, there's no need to have the original `Value`. What do you think @HalidOdat ?

Reply:
> I think maybe this method should instead be an associated function of the `JSON` struct, the `obj` is enough to get it's fields, there's no need to have the original `Value`. What do you think @HalidOdat Its?

It's better if we move this to `GcObject`, so we can take a &self

Reply:
Done!